Mancala.hs

```diff
@@ -11,18 +11,18 @@ type Board = (Side, Side)  -- TODO: not sure if using just a list is any better
 type InternalState = Board 
 data Action = Action Int -- index of a selected pocket (non-empty)
          deriving (Ord,Eq)
-data State = State InternalState ([Action], [Action])  
+data State = State InternalState ([Action], [Action])
          deriving (Ord, Eq, Show)
 
 data Result = EndOfGame Double State    -- end of game: value, starting state
             | EndOfTurn State        -- the other player continues 
-            | ContinueTurn State        -- current player takes another turn
+            | ContinueTurn State        -- current player takes another turn -- might not be necessary. can just call person_play or opponent_play again
```

Comment:
Yea that works too. It's essentially a question of do we want to push the logic to person_play/opponent_play OR should it be part of the Game type. I prefer keeping this for two reasons:
1. Ideally Play.hs should not know the internals of the game. It should just accpet a game result and move forward.
2. Since we are probably gonna implement some sort of minimax bot, keeping this state would make it easier to implement (no need to use if else just one more pattern match I imagine)

Reply:
Yeah, that would make sense - just more lines of code to deal with haha. We'll go with this implementation then!